### ISSUE TITLE
Linux/Mac Mods Still Broken due to Case Sensitivty

### ISSUE THREAD

**Issue by hubismal** [Regalis11/Barotrauma#4692]
[high prio????]
i've already made this issue before and it was falsely closed, but I can't find the old ticket.

linux filepath references are case sensitive, windows isn't case sensitive. this leads to mod creators (who use windows) to just ignore case sensitivity, and then everything breaks on linux and mac. so this leads to invisible monsters (can't find wrong cased png files) and crashing while trying to spectate (can't find wrong cased ragdoll files) and so on.

basically linux/mac people can't join windows modded servers and play which is kind of a big deal

ideally windows should be forced to have case sensitivity enabled. or vice versa.

**Comment by 3e849f2e5c** [comment]
Additionally I've noticed that a trailing whitespace in filelist.xml paths will make the mod incompatible with Linux

**Comment by Regalis11** [comment]
Todo note: the ToolBox.CleanUpPathCrossPlatform method was supposed to handle this, but looks like it isn't working correctly or being used in all the places it should.

**Comment by 3e849f2e5c** [comment]
I've fixes the mentioned cases in the `feature/linux-mod-compatibility` branch, needs to be tested that it works correctly on Windows just in case.

I hacked together [this mod](https://app.zenhub.com/files/93301055/a71f02f3-5102-4a1b-9986-d776bf12dfea/download) to test the changes that has file case inconsistencies everywhere and the mod works correctly on Linux.

**Comment by 3e849f2e5c** [comment]
@hubismal if you had a list of some mods on the workshop that do not work on Linux due to case issues I'd like to double check if my changes have fixed them.

**Comment by 3e849f2e5c** [comment]
Mods tested

1. [Dead Space](https://steamcommunity.com/sharedfiles/filedetails/?id=1863840119)
    - Fixed: Some monster had sprites and ragdolls that failed to load
    - Wontfix: French localization has broken XML, this is the mods fault
2. [Improved Husks](https://steamcommunity.com/sharedfiles/filedetails/?id=2085783214)
    - Fixed: Some characters had sprites that failed to load
3. [Colonial Marines](https://steamcommunity.com/sharedfiles/filedetails/?id=1803297935)
    - Fixed: Would not download from workshop on Linux
    - Wontfix: Job gear has incorrect texture paths, this is the mods fault

What needs to be tested:
- Ensure that the hashes match between Windows, macOS Linux and that they can join modded servers without issues.
- Check that this change didn't break anything on Windows or macOS, I've only tested the changes on Linux.

**Comment by Rokvach** [comment]
Tested the listed mods on Windows and didn't encounter any issues
Tested joining a modded server with the listed mods both ways with Markus, Linux => Windows server and Windows => Linux server and no issues.

Mac hasn't been tested.

Ready for code review and a merge.

**Comment by Rokvach** [comment]
Closing as this is just waiting for a merge
